syntax = "proto3";
package related_resource;




import "resource_monitor_sdk/model/resource_monitor/related_resource.proto";

/*
ListRelatedResource请求
*/
message ListRelatedResourceRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 pageSize = 2 ;
    /*
    所属模型id
    */
    string objectId = 3 ;
    /*
    关联名称
    */
    string name = 4 ;
}

/*
ListRelatedResource返回
*/
message ListRelatedResourceResponse {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 pageSize = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated resource_monitor.RelatedResource list = 4 ;
}

/*
ListRelatedResourceApi返回
*/
message ListRelatedResourceResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    related_resource.ListRelatedResourceResponse data = 4 ;
}

